Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/NodeFactory.php

```diff
@@ -57,13 +57,17 @@ public function __construct(
 
     /**
      * @param array<string,string> $nodeRow Node Row from projection (<prefix>_node table)
-     * @throws NodeTypeNotFoundException
      */
     public function mapNodeRowToNode(
         array $nodeRow,
         DimensionSpacePoint $dimensionSpacePoint,
         VisibilityConstraints $visibilityConstraints
     ): Node {
+        try {
+            $nodeType = $this->nodeTypeManager->getNodeType($nodeRow['nodetypename']);
+        } catch (NodeTypeNotFoundException $exception) {
+            $nodeType = null;
+        }
```

Comment:
Hmm im not sure i like to use exceptions as control flow here (exceptions are always expensive - one could argue that this is an edgecase and not the path taken in 90% of the cases ... so well fine as well)

But how about

A: adding a new method wich is nullable? (Or make `getNodeType` nullable -> no i think rather not?)

```suggestion
        $nodeType = $this->nodeTypeManager->findNodeType($nodeRow['nodetypename']);
```

B: ask if the nt mngr "has" the node:

```suggestion
        $nodeType = $this->nodeTypeManager->hasNodeType($nodeRow['nodetypename'])
            ? $this->nodeTypeManager->findNodeType($nodeRow['nodetypename'])
            : null;
```

Reply:
@nezaniel agrees ^^

Reply:
has been adjusted where available